### PR TITLE
Add validation for at least one image tag.

### LINKED ISSUE #9: Required Tag Association
#### As a user I want all images to have at least one tag.

__Story__:

The tag feature has been a hit. Our users love being able to filter images by
cats, SR-71s, and "AppFolio tshirts" to name a few. In fact, we have discovered
that images without any tags receive significantly less views than those with
tags. As a result, we have decided to require all images to have at least one
tag.

__Acceptance criteria__:
- [x] I cannot save an image without adding at least one tag.
- [x] An error message is associated with the appropriate input field when a
tag is not provided.
- [ ] All images have at least one tag.

__Discussion Topic__:
- [ ] What do you do about existing images that have no tags?

__Dependencies__:
- Image Tags
- Refactor flow tests to use page objects

### PR DESCRIPTION
**Pull request by lcguang**
Closes #9

### CODE REVIEW

File: test/models/image_test.rb [hunk 1]
```diff
@@ -66,6 +66,12 @@ def test_tag_list__failure__tag_too_long
     assert_equal image.errors.messages[:tag_list].first, 'Plz keep each tag to <20 characters.'
   end
 
+  def test_tag_list__failure__no_tags
+    image = Image.new(url: 'http://a.com/image.jpg', tag_list: '')
+    refute_predicate image, :valid?
+    assert_equal image.errors.messages[:tag_list].first, 'Must have at least one tag'
```
Comment (pololee) [inline test/models/image_test.rb]: `assert_equal <what you expect>, <what the actual value is>`.
So the order of the arguments is wrong here.

File: db/migrate/20180327001008_add_dummy_tags_to_images.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class AddDummyTagsToImages < ActiveRecord::Migration[5.1]
+  def up
+    execute("insert into tags (name, taggings_count) values ('dummy', (select count(*) from images))")
```
Comment (pololee) [inline db/migrate/20180327001008_add_dummy_tags_to_images.rb]: I don't think this is the correct migration. We only want to add `dummy` to images which don't have any tags. But this migration will add `dummy` to all the existing images.

File: db/migrate/20180327001008_add_dummy_tags_to_images.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    execute("insert into taggings (tag_id, taggable_type, taggable_id, context, created_at)
+      select ( select t.id from tags t where name = 'dummy' ), 'Image', i.id, 'tags',  '2018-03-27 12:34:56'
```
Comment (pololee) [inline db/migrate/20180327001008_add_dummy_tags_to_images.rb]: I would recommending not hard coding timestamp, but use the actual time when this migration runs.

File: db/migrate/20180327001008_add_dummy_tags_to_images.rb (hunk 2 above)
Comment (lcguang) [reply db/migrate/20180327001008_add_dummy_tags_to_images.rb]: Fixed.

File: db/migrate/20180327001008_add_dummy_tags_to_images.rb (hunk 3 above)
Comment (lcguang) [reply db/migrate/20180327001008_add_dummy_tags_to_images.rb]: We're using `datetime()` for sqlite3. Should we make our migration file more flexible by adding `if` / `else` statements for different database types?

Comment (pololee) [reply db/migrate/20180327001008_add_dummy_tags_to_images.rb]: This is good enough. No need to worry about different DB support.

**Pull request merged**